app/src/main/java/org/literacyapp/service/FaceRecognitionTrainingJobService.java

```diff
@@ -16,6 +16,7 @@ public boolean onStartJob(JobParameters jobParameters) {
         Log.i(getClass().getName(), "onStartJob");
         TrainingHelper trainingHelper = new TrainingHelper(getApplicationContext());
         trainingHelper.extractFeatures();
```

Comment:
I guess the synchronized keyword should prevent them from getting mixed up. But it also wouldn't be a huge problem as the service runs every 15 or 30 minutes and if the trainClassifier was executed too early it gets the new StudentImageFeatures the next time